### ISSUE TITLE
Add support for exclude regex.

### ISSUE THREAD

**Issue by gavenkoa** [ggreer/the_silver_searcher#1244]
I want to cleanup output by excluding matches that satisfy certain regexes.

I can:

    ag REGEX | grep -v IGNORE

but I'd like CLI option in `ag` to handle such cases as above command useless if I request context, like:

    ag -C10 REGEX

**Comment by genivia-inc** [comment]
Indeed, this is very handy to have. This feature is implemented by [ugrep's "negative patterns"](https://github.com/Genivia/ugrep#not) that are specified with `-N PATTERN`. Negative patterns can be used to "zap" parts of the input that should not be searched, for example comment blocks in source code files when searching for patterns that should not match anything in comment blocks.
